### DISCUSSION TITLE
What is `Stream.ObjectInvariant()`?

### DISCUSSION THREAD

**Discussion by colejohnson66** [dotnet/runtime#101155] [General] [answered]
There's an obsoleted protected method on `Stream`, `ObjectInvariant()`:

https://github.com/dotnet/runtime/blob/e6f1fd829709e64ae91c2a389581d2169f0bd9a8/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs#L944-L945

What was its purpose? The [documentation says](https://learn.microsoft.com/en-us/dotnet/api/system.io.stream.objectinvariant?view=net-8.0) that it serves to "provide[s] support for a Contract", but how? [`Contract.Invariant`](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.contracts.contract.invariant?view=net-8.0) wants a boolean but this method returns nothing.

**Comment by elgonzo** [comment, marked as answer]
See here for some more info on how this kind of method was supposed to be used/implemented in the past: https://learn.microsoft.com/en-us/dotnet/framework/debug-trace-profile/code-contracts#invariants
